### DISCUSSION TITLE
version value must be string

### DISCUSSION THREAD

**Discussion by srimanivinay97** [Homebrew/discussions#973] [Getting started]
/usr/local/Homebrew/Library/Homebrew/version.rb:368:in `initialize': Version value must be a string; got a NilClass () (TypeError)

I am using latest macOS version Big Sur 11.2.2

I think this problem is solved . I know that all Big Sur version faced this issue.

I removed old version of Xcode ctl and installed new Xcode ctl  and run the brew update 2 times 
but still I am facing this issue pls some one help me guys.  
![Screenshot 2021-03-02 at 6 08 29 PM](https://user-images.githubusercontent.com/42007271/109650038-ae969080-7b82-11eb-87d4-baa8524d14ed.png)

**Comment by carlocab** [comment]
Does `brew config` and `brew doctor` work? If so, what do they say?

**Reply by srimanivinay97** [reply]
same result as shown in image.

**Comment by shoesforindustry** [comment]
I have this as well. Just running brew gives: 
`brew
/usr/local/Homebrew/Library/Homebrew/version.rb:198:in `initialize': Version value must be a string; got a NilClass () (TypeError)
	from /usr/local/Homebrew/Library/Homebrew/os/mac/version.rb:26:in `initialize'
......`
Monteray 12.4 Imac 4 GHz Quad-Core Intel Core i7

Any ideas

**Reply by gromgit** [reply]
Try running `brew update-reset`. Does that throw any errors?

**Reply by shoesforindustry** [reply]
brew update-reset produces a list of unlinked 
`Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /usr/local/Frameworks/Python.framework/Headers
  /usr/local/Frameworks/Python.framework/Python
....
`
`brew cleanup`
this runs OK until
`Error: Permission denied @ rb_sysopen - /usr/local/var/homebrew/locks/update`

Let me know if you need anything else?

**Reply by gromgit** [reply]
What does `brew config` and `brew doctor` output now?

**Reply by shoesforindustry** [reply]
`brew config`
HOMEBREW_VERSION: 3.5.2-33-g79afa01
ORIGIN: https://github.com/Homebrew/brew
HEAD: 79afa0177bbff33aa0764a1a7c9acb8f64c743b5
Last commit: 17 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ee0086c2bed6a2f06df5233d8373d95c820c4e81
Core tap last commit: 18 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 13.1.6 build 1316
Git: 2.32.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.4-x86_64
CLT: 13.4.0.0.1.1651278267
Xcode: 13.4.1

**Reply by shoesforindustry** [reply]
brew cleanup fails as above same error message.
Thanks for trying to help gromgit, appreciated, is there anything else I can do?

**Reply by gromgit** [reply]
As I asked previously, what's the output of `brew doctor`?

**Reply by shoesforindustry** [reply]
This seemed to fix the brew cleanup
`sudo chmod a+w /usr/local/var/homebrew/locks/update`

brew doctor looks good, but of course I don't have anything installed in brew now...

**Reply by shoesforindustry** [reply]
After the above - Brew doctor is now ready to brew and I am now re-installing apps

**Reply by Dwebtron** [reply]
> This seemed to fix the brew cleanup `sudo chmod a+w /usr/local/var/homebrew/locks/update`
> 
> brew doctor looks good, but of course I don't have anything installed in brew now...

I came here for the same issue and this returns : 
"chmod: /usr/local/var/homebrew/locks/update: No such file or directory"

**Reply by gromgit** [reply]
Note that there are currently _three_ different official Homebrew roots:
* `/usr/local` on Intel macOS
* `/opt/homebrew` on ARM64 macOS
* `/home/linuxbrew/.linuxbrew` on Linux

and there are a whole bunch of folks who choose to install Homebrew in an infinite number of unofficial locations for their own reasons.

Whenever you see anyone specify a fully-qualified Homebrew path like `/usr/local/var/homebrew/locks/update`, you should _always_ use `brew --prefix` to substitute the correct Hombrew root path for _your_ system. In the above case, the correct command for _everyone_, regardless of where their Homebrew is installed, is:
```
sudo chmod a+w $(brew --prefix)/var/homebrew/locks/update
```

**Reply by Dwebtron** [reply]
> Note that there are currently _three_ different official Homebrew roots:
> 
>     * `/usr/local` on Intel macOS
> 
>     * `/opt/homebrew` on ARM64 macOS
> 
>     * `/home/linuxbrew/.linuxbrew` on Linux
> 
> 
> and there are a whole bunch of folks who choose to install Homebrew in an infinite number of unofficial locations for their own reasons.
> 
> Whenever you see anyone specify a fully-qualified Homebrew path like `/usr/local/var/homebrew/locks/update`, you should _always_ use `brew --prefix` to substitute the correct Hombrew root path for _your_ system. In the above case, the correct command for _everyone_, regardless of where their Homebrew is installed, is:
> 
> ```
> sudo chmod a+w $(brew --prefix)/var/homebrew/locks/update
> ```

Having the exact same problem on an M1 mac running 14.6.1 (23G93). Tried 

>  chmod a+w /opt/homebrew/var/homebrew/locks/update   

which seemed to work, but brew doctor still returns
`/usr/local/Library/Homebrew/version.rb:186:in `initialize': Version value must be a string (TypeError)
	from /usr/local/Library/Homebrew/os/mac/version.rb:25:in `initialize'
	from /usr/local/Library/Homebrew/os/mac.rb:16:in `new'
	from /usr/local/Library/Homebrew/os/mac.rb:16:in `version'
	from /usr/local/Library/Homebrew/os.rb:16:in `<module:OS>'
	from /usr/local/Library/Homebrew/os.rb:1:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/Library/Homebrew/global.rb:7:in `<top (required)>'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/Library/brew.rb:15:in `<main>'
`

I've already tried fully uninstalling/reinstalling. I don't know what else to do, I've been unable to solve this problem for months.
